### PR TITLE
[release-1.23] feat: support workload identity setting in static PV mount

### PR DESCRIPTION
**Pull request by cvvz**
This is a cherry-pick of #1204

**Here are the csi driver's logs:**

* When mount static PV, first we skip `NodeStageVolume`, then we use service account token to get the account key during `NodePublishVolume`
```
csi-blob-node-859kb blob I0108 12:03:59.554570   25337 utils.go:103] GRPC call: /csi.v1.Node/NodeStageVolume
csi-blob-node-859kb blob I0108 12:03:59.554600   25337 utils.go:104] GRPC request: {"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/e53ce882ae47f30739a21cc04a37ac4a14a19894e1551e7d78909e7e1d52d167/globalmount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["-o allow_other","--file-cache-timeout-in-seconds=120"]}},"access_mode":{"mode":5}},"volume_context":{"clientID":"f696d967-7475-4661-9dcf-9e8dc882b638","containerName":"weizhichencontainer","resourcegroup":"weizhichen-playground","storageaccount":"weizhichenaccount"},"volume_id":"unique_volume_id"}
csi-blob-node-859kb blob I0108 12:03:59.554743   25337 nodeserver.go:256] Skip NodeStageVolume for volume(unique_volume_id) since clientID f696d967-7475-4661-9dcf-9e8dc882b638 is provided but service account token is empty
csi-blob-node-859kb blob I0108 12:03:59.554752   25337 utils.go:110] GRPC response: {}
csi-blob-node-859kb blob I0108 12:03:59.592384   25337 utils.go:103] GRPC call: /csi.v1.Node/NodePublishVolume
csi-blob-node-859kb blob I0108 12:03:59.592409   25337 utils.go:104] GRPC request: {"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/e53ce882ae47f30739a21cc04a37ac4a14a19894e1551e7d78909e7e1d52d167/globalmount","target_path":"/var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["-o allow_other","--file-cache-timeout-in-seconds=120"]}},"access_mode":{"mode":5}},"volume_context":{"clientID":"f696d967-7475-4661-9dcf-9e8dc882b638","containerName":"weizhichencontainer","csi.storage.k8s.io/ephemeral":"false","csi.storage.k8s.io/pod.name":"statefulset-blob-0","csi.storage.k8s.io/pod.namespace":"default","csi.storage.k8s.io/pod.uid":"06136265-f1c7-499c-9443-f52ca6543b19","csi.storage.k8s.io/serviceAccount.name":"saname","csi.storage.k8s.io/serviceAccount.tokens":"{\"api://AzureADTokenExchange\":{\"token\":\"eyJhbGciOiJSUzI1NiIsImtpZCI6IjBPbXNQd0ZBeklUb095aFF6eV9oaXI0UE5FbDlqQ3ZwR19xUkc5ZEhpNWsifQ.eyJhdWQiOlsiYXBpOi8vQXp1cmVBRFRva2VuRXhjaGFuZ2UiXSwiZXhwIjoxNzA0NzE5MDM5LCJpYXQiOjE3MDQ3MTU0MzksImlzcyI6Imh0dHBzOi8vZWFzdHVzLm9pYy5wcm9kLWFrcy5henVyZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3LzMyOWVmZTZmLWU0MDctNGVhZi04YTc0LTFmYTk4NzIyYjRjMy8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImRlZmF1bHQiLCJwb2QiOnsibmFtZSI6InN0YXRlZnVsc2V0LWJsb2ItMCIsInVpZCI6IjA2MTM2MjY1LWYxYzctNDk5Yy05NDQzLWY1MmNhNjU0M2IxOSJ9LCJzZXJ2aWNlYWNjb3VudCI6eyJuYW1lIjoic2FuYW1lIiwidWlkIjoiYTBjMjY5YjUtNjk4Zi00NmNiLWFjYTYtZTE2MTUwMzE4NGU2In19LCJuYmYiOjE3MDQ3MTU0MzksInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OnNhbmFtZSJ9.egl1eRdJ_Rj5DXPpkBxTFDw3rnD5pqclsgb50HTGMzcWdWYcBggzc4NbvFbWVmRYWPwb6KPpVGPSuet6X9YT9iNSGvarWjJd-ytw7ikP_pNysKgKBbBDbgaFlJnTMl37IwCC3oR-Kyt2ePCM5Y1GBDJOLgQxuHlTrR-yURSXvi6Y2PicV7P0udZcchsUK3jxGLI8sG_EAmBgirVlIG5oOZ6imN8DS2dp3ux72YCuZNviK0Xp3QQNLN1G7bd_D7ilkLv8-ohDgwCqXiw88_Fp0W2DMufAMmobiYoYaUnMJY1wXkdCf_0ROMdJEbxttEXVn40npNHRupduTFSBOARHkCL7NHYMxyJCQyK5UthRZLUF-rc-m_FaXnJmZmqo7nbfi1pEjTc9pyVd_HaEtr5yT5Gvrr7Bwkucgm68ISxftLyisJXyd7y1ITZUGJujSTMJv_DSbuKzZExX5uRvZ322vWHpYGaGrrafW7pAWX54rH_H2rdjNG12JAecYdyUOVTR2KUfcKVoW9dKcCsHnHNXgEzM9Hlwbx0DiqjEhxnBU_I5I7g39L5e3Yic5k279Oe0Bsy2QR5LAMX3MLrn16u11oFysc0BTUAry3Lr-oRuL3GFnUXvn4RoV4wiO_DXupxeUncBhnVhuSeLBjP6TqDGmx_pZrzwoGfZNwlMxH9dRWg\",\"expirationTimestamp\":\"2024-01-08T13:03:59Z\"}}","resourcegroup":"weizhichen-playground","storageaccount":"weizhichenaccount"},"volume_id":"unique_volume_id"}
csi-blob-node-859kb blob I0108 12:03:59.592605   25337 nodeserver.go:85] NodePublishVolume: volume(unique_volume_id) mount on /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount with service account token, clientID: f696d967-7475-4661-9dcf-9e8dc882b638
csi-blob-node-859kb blob I0108 12:03:59.592931   25337 blob.go:416] parsing volumeID(unique_volume_id) return with error: error parsing volume id: "unique_volume_id", should at least contain two #
csi-blob-node-859kb blob I0108 12:03:59.592977   25337 blob.go:497] volumeID(unique_volume_id) authEnv: []
csi-blob-node-859kb blob I0108 12:03:59.592998   25337 blob.go:522] clientID(f696d967-7475-4661-9dcf-9e8dc882b638) is specified, use service account token to get account key
csi-blob-node-859kb blob I0108 12:04:00.451454   25337 nodeserver.go:385] target /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount
csi-blob-node-859kb blob protocol
csi-blob-node-859kb blob
csi-blob-node-859kb blob volumeId unique_volume_id
csi-blob-node-859kb blob context map[clientID:f696d967-7475-4661-9dcf-9e8dc882b638 containerName:weizhichencontainer csi.storage.k8s.io/ephemeral:false csi.storage.k8s.io/pod.name:statefulset-blob-0 csi.storage.k8s.io/pod.namespace:default csi.storage.k8s.io/pod.uid:06136265-f1c7-499c-9443-f52ca6543b19 csi.storage.k8s.io/serviceAccount.name:saname csi.storage.k8s.io/serviceAccount.tokens:{"api://AzureADTokenExchange":{"token":"eyJhbGciOiJSUzI1NiIsImtpZCI6IjBPbXNQd0ZBeklUb095aFF6eV9oaXI0UE5FbDlqQ3ZwR19xUkc5ZEhpNWsifQ.eyJhdWQiOlsiYXBpOi8vQXp1cmVBRFRva2VuRXhjaGFuZ2UiXSwiZXhwIjoxNzA0NzE5MDM5LCJpYXQiOjE3MDQ3MTU0MzksImlzcyI6Imh0dHBzOi8vZWFzdHVzLm9pYy5wcm9kLWFrcy5henVyZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3LzMyOWVmZTZmLWU0MDctNGVhZi04YTc0LTFmYTk4NzIyYjRjMy8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImRlZmF1bHQiLCJwb2QiOnsibmFtZSI6InN0YXRlZnVsc2V0LWJsb2ItMCIsInVpZCI6IjA2MTM2MjY1LWYxYzctNDk5Yy05NDQzLWY1MmNhNjU0M2IxOSJ9LCJzZXJ2aWNlYWNjb3VudCI6eyJuYW1lIjoic2FuYW1lIiwidWlkIjoiYTBjMjY5YjUtNjk4Zi00NmNiLWFjYTYtZTE2MTUwMzE4NGU2In19LCJuYmYiOjE3MDQ3MTU0MzksInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpkZWZhdWx0OnNhbmFtZSJ9.egl1eRdJ_Rj5DXPpkBxTFDw3rnD5pqclsgb50HTGMzcWdWYcBggzc4NbvFbWVmRYWPwb6KPpVGPSuet6X9YT9iNSGvarWjJd-ytw7ikP_pNysKgKBbBDbgaFlJnTMl37IwCC3oR-Kyt2ePCM5Y1GBDJOLgQxuHlTrR-yURSXvi6Y2PicV7P0udZcchsUK3jxGLI8sG_EAmBgirVlIG5oOZ6imN8DS2dp3ux72YCuZNviK0Xp3QQNLN1G7bd_D7ilkLv8-ohDgwCqXiw88_Fp0W2DMufAMmobiYoYaUnMJY1wXkdCf_0ROMdJEbxttEXVn40npNHRupduTFSBOARHkCL7NHYMxyJCQyK5UthRZLUF-rc-m_FaXnJmZmqo7nbfi1pEjTc9pyVd_HaEtr5yT5Gvrr7Bwkucgm68ISxftLyisJXyd7y1ITZUGJujSTMJv_DSbuKzZExX5uRvZ322vWHpYGaGrrafW7pAWX54rH_H2rdjNG12JAecYdyUOVTR2KUfcKVoW9dKcCsHnHNXgEzM9Hlwbx0DiqjEhxnBU_I5I7g39L5e3Yic5k279Oe0Bsy2QR5LAMX3MLrn16u11oFysc0BTUAry3Lr-oRuL3GFnUXvn4RoV4wiO_DXupxeUncBhnVhuSeLBjP6TqDGmx_pZrzwoGfZNwlMxH9dRWg","expirationTimestamp":"2024-01-08T13:03:59Z"}} resourcegroup:weizhichen-playground storageaccount:weizhichenaccount]
csi-blob-node-859kb blob mountflags [-o allow_other --file-cache-timeout-in-seconds=120]
csi-blob-node-859kb blob mountOptions [-o allow_other --file-cache-timeout-in-seconds=120 --pre-mount-validate=true --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/unique_volume_id --container-name=weizhichencontainer]
csi-blob-node-859kb blob args /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount -o allow_other --file-cache-timeout-in-seconds=120 --pre-mount-validate=true --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/unique_volume_id --container-name=weizhichencontainer
csi-blob-node-859kb blob serverAddress weizhichenaccount.blob.core.windows.net
csi-blob-node-859kb blob I0108 12:04:00.451518   25337 nodeserver.go:167] start connecting to blobfuse proxy, protocol: , args: /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount -o allow_other --file-cache-timeout-in-seconds=120 --pre-mount-validate=true --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/unique_volume_id --container-name=weizhichencontainer
csi-blob-node-859kb blob I0108 12:04:00.452588   25337 nodeserver.go:176] begin to mount with blobfuse proxy, protocol: , args: /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount -o allow_other --file-cache-timeout-in-seconds=120 --pre-mount-validate=true --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/unique_volume_id --container-name=weizhichencontainer
csi-blob-node-859kb blob I0108 12:04:00.681484   25337 mount_linux.go:282] Detected umount with safe 'not mounted' behavior
csi-blob-node-859kb blob I0108 12:04:00.681757   25337 nodeserver.go:644] blobfuse mount at /var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount success
csi-blob-node-859kb blob I0108 12:04:00.681826   25337 nodeserver.go:443] volume(unique_volume_id) mount on "/var/lib/kubelet/pods/06136265-f1c7-499c-9443-f52ca6543b19/volumes/kubernetes.io~csi/pv-blob/mount" succeeded
csi-blob-node-859kb blob I0108 12:04:00.681863   25337 utils.go:110] GRPC response: {}
```

* When mount ephemeral volume, we use service account token to get the account key during `NodePublishVolume`
```
csi-blob-node-859kb blob I0108 12:07:27.960774   25337 utils.go:103] GRPC call: /csi.v1.Node/NodePublishVolume
csi-blob-node-859kb blob I0108 12:07:27.960800   25337 utils.go:104] GRPC request: {"target_path":"/var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":7}},"volume_context":{"clientID":"f696d967-7475-4661-9dcf-9e8dc882b638","containerName":"weizhichencontainer","csi.storage.k8s.io/ephemeral":"true","csi.storage.k8s.io/pod.name":"nginx-blobfuse-inline-volume","csi.storage.k8s.io/pod.namespace":"default","csi.storage.k8s.io/pod.uid":"35714371-f544-4ae5-b14b-83c1d3791711","csi.storage.k8s.io/serviceAccount.name":"saname","csi.storage.k8s.io/serviceAccount.tokens":"{\"api://AzureADTokenExchange\":{\"token\":\"eyJhbGciOiJSUzI1NiIsImtpZCI6IjBPbXNQd0ZBeklUb095aFF6eV9oaXI0UE5FbDlqQ3ZwR19xUkc5ZEhpNWsifQ.eyJhdWQiOlsiYXBpOi8vQXp1cmVBRFRva2VuRXhjaGFuZ2UiXSwiZXhwIjoxNzA0NzE5MjQ3LCJpYXQiOjE3MDQ3MTU2NDcsImlzcyI6Imh0dHBzOi8vZWFzdHVzLm9pYy5wcm9kLWFrcy5henVyZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3LzMyOWVmZTZmLWU0MDctNGVhZi04YTc0LTFmYTk4NzIyYjRjMy8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImRlZmF1bHQiLCJwb2QiOnsibmFtZSI6Im5naW54LWJsb2JmdXNlLWlubGluZS12b2x1bWUiLCJ1aWQiOiIzNTcxNDM3MS1mNTQ0LTRhZTUtYjE0Yi04M2MxZDM3OTE3MTEifSwic2VydmljZWFjY291bnQiOnsibmFtZSI6InNhbmFtZSIsInVpZCI6ImEwYzI2OWI1LTY5OGYtNDZjYi1hY2E2LWUxNjE1MDMxODRlNiJ9fSwibmJmIjoxNzA0NzE1NjQ3LCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6ZGVmYXVsdDpzYW5hbWUifQ.c5yCs55hCIne4XivHOvqi4EebpjNODScw8eEsAQoky9EiDaICJv-HbdsjuiIM2RYkohXuQDfp7RvwdFXAgcFYBskfJ4UMBK8NgcD3AbeyACVS3xf10QvGvKDeRZfX9mA8TIdKneEZj2c5KIJ_wzQOeRBAQS4XFy2pN897rSm-j6zgfa6DSc9ItZ3VhoNiAb68XBuYL230i6DcSv11cJgobTHuLfwL4dsPNka3wwC4MqqQBjBoBzpwgYhACyGcyaIACvMSRltRwN8uXhKLz3i_P3YjoQUIxxfNjqORuOYqDj3OC-Ww1MeKfqp8_UIpFsE8fv1UJog_suaRKoj4snvxnaZ6Mie1W-TKsuaq9qXTTfRIr44IEa80gcVX9bbSJkyAhg71KxqnFgCmoTDbsI4yMYaZXNYt_A7iG3eHkTG9e7g2ZD0GQHc8qfC-BtK-Lo0QACGfqmTyh8Ke4uznziAh9npotn9kFzMyk74miect80l7JGoTFxtdogFzd-HJ04n72dYDKvrpheWb9QW4DFmsxt8vOFvEZ4kxDrVE9N_7JCucmkyvOHXr3mekLqO4berpSaw8hIDGDcqzLpK2mqglrVXyH0SXRgB4aK7f6vsGUcDToKBVjHvLVvh04gKasWVUv5xkoRYVLBZoGW1grhyaPPUmzX_Im_pdzD4CT-RtHI\",\"expirationTimestamp\":\"2024-01-08T13:07:27Z\"}}","resourcegroup":"weizhichen-playground","storageaccount":"weizhichenaccount"},"volume_id":"csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127"}
csi-blob-node-859kb blob I0108 12:07:27.960933   25337 nodeserver.go:85] NodePublishVolume: volume(csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127) mount on /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount with service account token, clientID: f696d967-7475-4661-9dcf-9e8dc882b638
csi-blob-node-859kb blob I0108 12:07:27.961330   25337 blob.go:416] parsing volumeID(csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127) return with error: error parsing volume id: "csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127", should at least contain two #
csi-blob-node-859kb blob I0108 12:07:27.961356   25337 blob.go:497] volumeID(csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127) authEnv: []
csi-blob-node-859kb blob I0108 12:07:27.961370   25337 blob.go:522] clientID(f696d967-7475-4661-9dcf-9e8dc882b638) is specified, use service account token to get account key
csi-blob-node-859kb blob I0108 12:07:28.867295   25337 nodeserver.go:385] target /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount
csi-blob-node-859kb blob protocol
csi-blob-node-859kb blob
csi-blob-node-859kb blob volumeId csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127
csi-blob-node-859kb blob context map[clientID:f696d967-7475-4661-9dcf-9e8dc882b638 containerName:weizhichencontainer csi.storage.k8s.io/ephemeral:true csi.storage.k8s.io/pod.name:nginx-blobfuse-inline-volume csi.storage.k8s.io/pod.namespace:default csi.storage.k8s.io/pod.uid:35714371-f544-4ae5-b14b-83c1d3791711 csi.storage.k8s.io/serviceAccount.name:saname csi.storage.k8s.io/serviceAccount.tokens:{"api://AzureADTokenExchange":{"token":"eyJhbGciOiJSUzI1NiIsImtpZCI6IjBPbXNQd0ZBeklUb095aFF6eV9oaXI0UE5FbDlqQ3ZwR19xUkc5ZEhpNWsifQ.eyJhdWQiOlsiYXBpOi8vQXp1cmVBRFRva2VuRXhjaGFuZ2UiXSwiZXhwIjoxNzA0NzE5MjQ3LCJpYXQiOjE3MDQ3MTU2NDcsImlzcyI6Imh0dHBzOi8vZWFzdHVzLm9pYy5wcm9kLWFrcy5henVyZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3LzMyOWVmZTZmLWU0MDctNGVhZi04YTc0LTFmYTk4NzIyYjRjMy8iLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImRlZmF1bHQiLCJwb2QiOnsibmFtZSI6Im5naW54LWJsb2JmdXNlLWlubGluZS12b2x1bWUiLCJ1aWQiOiIzNTcxNDM3MS1mNTQ0LTRhZTUtYjE0Yi04M2MxZDM3OTE3MTEifSwic2VydmljZWFjY291bnQiOnsibmFtZSI6InNhbmFtZSIsInVpZCI6ImEwYzI2OWI1LTY5OGYtNDZjYi1hY2E2LWUxNjE1MDMxODRlNiJ9fSwibmJmIjoxNzA0NzE1NjQ3LCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6ZGVmYXVsdDpzYW5hbWUifQ.c5yCs55hCIne4XivHOvqi4EebpjNODScw8eEsAQoky9EiDaICJv-HbdsjuiIM2RYkohXuQDfp7RvwdFXAgcFYBskfJ4UMBK8NgcD3AbeyACVS3xf10QvGvKDeRZfX9mA8TIdKneEZj2c5KIJ_wzQOeRBAQS4XFy2pN897rSm-j6zgfa6DSc9ItZ3VhoNiAb68XBuYL230i6DcSv11cJgobTHuLfwL4dsPNka3wwC4MqqQBjBoBzpwgYhACyGcyaIACvMSRltRwN8uXhKLz3i_P3YjoQUIxxfNjqORuOYqDj3OC-Ww1MeKfqp8_UIpFsE8fv1UJog_suaRKoj4snvxnaZ6Mie1W-TKsuaq9qXTTfRIr44IEa80gcVX9bbSJkyAhg71KxqnFgCmoTDbsI4yMYaZXNYt_A7iG3eHkTG9e7g2ZD0GQHc8qfC-BtK-Lo0QACGfqmTyh8Ke4uznziAh9npotn9kFzMyk74miect80l7JGoTFxtdogFzd-HJ04n72dYDKvrpheWb9QW4DFmsxt8vOFvEZ4kxDrVE9N_7JCucmkyvOHXr3mekLqO4berpSaw8hIDGDcqzLpK2mqglrVXyH0SXRgB4aK7f6vsGUcDToKBVjHvLVvh04gKasWVUv5xkoRYVLBZoGW1grhyaPPUmzX_Im_pdzD4CT-RtHI","expirationTimestamp":"2024-01-08T13:07:27Z"}} resourcegroup:weizhichen-playground storageaccount:weizhichenaccount]
csi-blob-node-859kb blob mountflags []
csi-blob-node-859kb blob mountOptions [ --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127 --container-name=weizhichencontainer --pre-mount-validate=true]
csi-blob-node-859kb blob args /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount  --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127 --container-name=weizhichencontainer --pre-mount-validate=true
csi-blob-node-859kb blob serverAddress weizhichenaccount.blob.core.windows.net
csi-blob-node-859kb blob I0108 12:07:28.867362   25337 nodeserver.go:167] start connecting to blobfuse proxy, protocol: , args: /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount  --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127 --container-name=weizhichencontainer --pre-mount-validate=true
csi-blob-node-859kb blob I0108 12:07:28.868732   25337 nodeserver.go:176] begin to mount with blobfuse proxy, protocol: , args: /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount  --use-https=true --cancel-list-on-mount-seconds=10 --empty-dir-check=false --tmp-path=/mnt/csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127 --container-name=weizhichencontainer --pre-mount-validate=true
csi-blob-node-859kb blob I0108 12:07:29.085735   25337 mount_linux.go:282] Detected umount with safe 'not mounted' behavior
csi-blob-node-859kb blob I0108 12:07:29.086029   25337 nodeserver.go:644] blobfuse mount at /var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount success
csi-blob-node-859kb blob I0108 12:07:29.086064   25337 nodeserver.go:443] volume(csi-faa9ea2d55003551aced3dcf1f6f293cf02c6096ea3ab82815112cc255b8b127) mount on "/var/lib/kubelet/pods/35714371-f544-4ae5-b14b-83c1d3791711/volumes/kubernetes.io~csi/persistent-storage/mount" succeeded
csi-blob-node-859kb blob I0108 12:07:29.086083   25337 utils.go:110] GRPC response: {}
```

### CODE REVIEW

File: pkg/blob/controllerserver_test.go [hunk 1]
```diff
@@ -164,18 +165,6 @@ func TestCreateVolume(t *testing.T) {
 		name     string
 		testFunc func(t *testing.T)
 	}{
-		{
```
Comment (andyzhangx) [inline pkg/blob/controllerserver_test.go:167]: why this ut is removed?

File: pkg/blob/controllerserver_test.go [hunk 2]
```diff
@@ -1321,21 +1296,6 @@ func TestControllerExpandVolume(t *testing.T) {
 				}
 			},
 		},
-		{
-			name: "invalid expand volume req",
```
Comment (andyzhangx) [inline pkg/blob/controllerserver_test.go:1325]: why this ut is removed?

File: pkg/blob/controllerserver_test.go (hunk 1 above)
Comment (cvvz) [reply pkg/blob/controllerserver_test.go:167]: The deleted uts are cherry-picks from https://github.com/kubernetes-sigs/blob-csi-driver/commit/ad9d27c27e3770e857a81a10d0e887b9f246a899#diff-0b174fc042381a284b07f142da447934352f84b42ab6fb5c3cc5082857b376b5

File: pkg/blob/controllerserver_test.go (hunk 2 above)
Comment (cvvz) [reply pkg/blob/controllerserver_test.go:1325]: ditto

**Comment by cvvz** [comment]
@andyzhangx Could you please take another look? The unit test coverage decreased was caused by the [ cherry-pick](https://github.com/kubernetes-sigs/blob-csi-driver/commit/ad9d27c27e3770e857a81a10d0e887b9f246a899#diff-0b174fc042381a284b07f142da447934352f84b42ab6fb5c3cc5082857b376b5), . And I've validated that the basic functions are normal.

**Comment by andyzhangx** [comment]
> @andyzhangx Could you please take another look? The unit test coverage decreased was caused by the [ cherry-pick](https://github.com/kubernetes-sigs/blob-csi-driver/commit/ad9d27c27e3770e857a81a10d0e887b9f246a899#diff-0b174fc042381a284b07f142da447934352f84b42ab6fb5c3cc5082857b376b5), . And I've validated that the basic function is normal.

@cvvz could you add ut back in master branch in another PR? I think we need to add those ut back. 
cc @MartinForReal

**Review by cvvz — COMMENTED** [review]
> @cvvz could you add ut back in master branch in another PR? I think we need to add those ut back. 
cc @MartinForReal  

Yes, That's what I'm thinking.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by cvvz** [comment]
> @cvvz could you add ut back in master branch in another PR? I think we need to add those ut back.
cc @MartinForReal

@andyzhangx  I've checked the code refactor and these ut are ok to be deleted, it won't affect the coverage. Actually, there is another ut that coverage no code, I've filed a PR #1214 to fix it.

**Pull request merged**